tuscan/compiler_wrapper.c

```diff
@@ -0,0 +1,48 @@
+/* Tuscan compiler wrapper
+ *
+ * Copyright 2016 Kareem Khazem. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file is a Jinja template that is filled out by
+ * tuscan/wrapper_gen.py. One such file will be generated for each
+ * native tool that will need to be redirected to a toolchain tool in
+ * accordance with the rules specified in
+ * toolchains/make_package/TOOLCHAIN/binutil_transforms.yaml.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+
+int main(int argc, char **argv){
+  int fd = mkstemp("/tmp/tuscan-native-XXXXXX");
+  if (fd == -1) {
+    perror("tuscan: mkstemp");
+    return 1;
+  }
+  dprintf(fd, "{{ native_program }}\n");
+  if (close(fd) == -1) {
+    perror("tuscan: close");
+    return 1;
+  }
+
+  char *toolchain_tool = "{{ toolchain_bin }}/{{ toolchain_program }}";
```

Comment:
I implemented all the changes you mentioned except for making this a `const char*`. This is because `new_argv` is required by execv to be of type `char * const[]` and not `const char * const[]`, and so assigning a `const char*` string to the array breaks the constness of that particular string.

`toolchain_tool` is also used as the _first_ parameter to `execv`, which does expect a `const char *`, but that's okay because it just gets upgraded to `const` when the function is invoked.